test/jasmine/tests/axes_test.js

```diff
@@ -4376,6 +4376,37 @@ describe('Test axes', function() {
             })
             .then(done, done.fail);
         });
+        it('should respect axis title placement on relayout', function(done) {
+
+            function getPos(gd, sel) {
+                return d3Select(gd).select(sel).node().getBoundingClientRect()
+            }
+
+            function assertLayout() {
+                var titleTop = getPos(gd, '.xtitle').top;
+                var tickBottom = getPos(gd, '.xtick').bottom;
+                expect(tickBottom).toBeLessThan(titleTop);
+            }
+            // TODO: This is failing now. 
+            // Is it maybe because there's overlap in these elements because of some padding?
+            // I'm also not sure that I've accessed the correct properties
+            var fig = require('@mocks/z-automargin-zoom.json');
+            
+            Plotly.newPlot(gd, fig)
+            console.log(d3Select(gd).select('.xtitle').node().getBoundingClientRect().top)
+            console.log(d3Select(gd).select('.xtitle').node().getBoundingClientRect().bottom)
```

Comment:
Let's remove these `console.log`s so that we enter `then` blocks.

Reply:
Yes :) My commit with this was a bit premature